client/store/product.js

```diff
@@ -84,6 +82,7 @@ const products = (state = initialState, action) => {
         filteredProducts: action.products
       }
     case FILTER_PRODUCTS:
+      //This reducer needs to be adjusted since categories are Category objects on the Products
       return {
         ...state,
         filteredProducts: state.allProducts.filter(product =>
```

Comment:
Good catch. So should line 89 just be...

product.categories.title.includes(action.category)

?

Reply:
So here is what I have for FILTER_PRODUCT in the reducer. I also had to make a small change in AllProducts component too. I can commit and submit this as a separate pull request.

```
return {
        ...state,
        filteredProducts: state.allProducts.filter(product => {
          const foundCategory = product.categories.find(
            currentCategory => action.category === currentCategory.title
          )
          return foundCategory || false
        })
```